rust/bridge/jni/src/util.rs

```diff
@@ -343,9 +354,17 @@ pub fn exception_check(env: &JNIEnv, fn_name: &'static str) -> Result<(), Signal
         let throwable = env.exception_occurred()?;
         env.exception_clear()?;
 
+        if let Some(exception_to_treat_as_null) = exception_to_treat_as_null {
+            if env.is_instance_of(throwable, exception_to_treat_as_null)? {
+                return Ok(JValue::Object(JObject::null()));
+            }
+        }
+
         let getmessage_sig = "()Ljava/lang/String;";
 
         let exn_type = exception_class_name(env, throwable).ok();
+        // Clear again in case we got an exception looking up the class name.
+        env.exception_clear()?;
```

Comment:
If there was an error looking up the class name, this will eat that error? Same as the one below.

I'd be tempted to suggest falling over, otherwise we don't know there's any problem.

Reply:
We're dropping the precise exception, but we'll get "(None)" or whatever in the output for the class name, and a dummy string (see below) for the message. Honestly I expect these to never occur; I don't know how I would even write a test case for it.